### DISCUSSION TITLE
Query data when using dask-awkward

### DISCUSSION THREAD

**Discussion by JuanBSLeite** [scikit-hep/uproot5#1283] [Q&A] [answered]
Dear experts,

I'm a starter on uproot.dask.  I'm trying to apply some selections on data using it.  Is there a way to apply selections to Dask-Awkward similar to Dask.dataframe.query?

Something like `ddf.query('y > x').compute()` ?

https://docs.dask.org/en/stable/generated/dask.dataframe.DataFrame.query.html

Thanks,
Juan

**Comment by jpivarski** [comment]
Yes, here's an example (given the array from [here](https://github.com/scikit-hep/awkward?tab=readme-ov-file#motivating-example)):

```python
>>> import awkward as ak
>>> import dask_awkward as dak
>>> array = ak.Array([
...     [{"x": 1.1, "y": [1]}, {"x": 2.2, "y": [1, 2]}, {"x": 3.3, "y": [1, 2, 3]}],
...     [],
...     [{"x": 4.4, "y": [1, 2, 3, 4]}, {"x": 5.5, "y": [1, 2, 3, 4, 5]}]
... ])

>>> delayed_array = dak.from_awkward(array, npartitions=1)
>>> selected_delayed_array = delayed_array[delayed_array.x > 3]

>>> selected_delayed_array.compute().show()
[[{x: 3.3, y: [1, 2, 3]}],
 [],
 [{x: 4.4, y: [1, 2, 3, 4]}, {x: 5.5, y: [1, ..., 5]}]]
```

Only the records in which `x > 3` remain, though the records with fields `x`, `y` are in variable-length lists and `x` and `y` have very different data types (scalar vs list).

**Reply by JuanBSLeite** [reply]
Thanks @jpivarski for the quick answer. But, I need to apply multiple selections using dataframe variables. Can I do selections like bellow?

`selected_delayed_array = delayed_array[delayed_array.x > 3 and delayed_array.y>1]`

**Reply by JuanBSLeite** [reply]
I found it where in the documentation. Thanks again!

**Reply by jpivarski** [reply, marked as answer]
Each `delayed_array.x > 3` and `delayed_array.y > 1` creates an array of booleans that can be used as a slice. To combine the two arrays of booleans with "and", use the bitwise operator `&` (with parentheses for correct order-of-operations):

```python
selected_delayed_array = delayed_array[(delayed_array.x > 3) & (delayed_array.y > 1)]
```

**Reply by jpivarski** [reply]
In this particular case, `x` field values are scalars and `y` field values are lists, which might broadcast when used together in the `&` operation, but I'd have to think a bit more.

Your case might not have this feature: if you have as many `x` values as you have `y` values, there would be no issue.
